### PR TITLE
Post-release updates for 1.15.x

### PR DESCRIPTION
**Pull request by malizz**
### Description
Post-release updates for 1.15.x
- update version 
- add changelog 1.15.3
- update go modules

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [x] not a security concern

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,14 +2,6 @@ module github.com/hashicorp/consul
 
 go 1.20
 
-replace (
-	github.com/hashicorp/consul/api => ./api
-	github.com/hashicorp/consul/envoyextensions => ./envoyextensions
-	github.com/hashicorp/consul/proto-public => ./proto-public
-	github.com/hashicorp/consul/sdk => ./sdk
-	github.com/hashicorp/consul/troubleshoot => ./troubleshoot
-)
```
Comment (zalimeni) [inline go.mod:11]: Was this removal intentional?

**Review by zalimeni — COMMENTED** [review]
Overall LGTM, one question in diff.

We might also want to change the PR title to reflect all the changes here, e.g. "
Post-release updates for 1.15.x"

File: go.mod (hunk 1 above)
Comment (malizz) [reply go.mod:11]: I can comment it out instead, I believe `go mod tidy` removed these lines. Thanks.

**Review by zalimeni — APPROVED** [review]

**Review by zalimeni — APPROVED** [review]

**Comment by zalimeni** [comment]
Squashed commits and removed `replace` comments for the running release branch.
